### PR TITLE
feat: cross file blocks

### PR DESCRIPTION
**Pull request by aidenybai**
**Please describe the changes this PR makes and why it should be merged:**

Supports this syntax:

```js
import { Button } from './different-file';

const ButtonBlock = block(Button); // cannot get the declaration
```

**Status**

- [ ] Code changes have been tested against prettier, or there are no code changes
- [ ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ ] This PR changes the codebase
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [ ] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/compiler/plugin.ts [hunk 1]
```diff
@@ -73,6 +106,26 @@ export const babelPlugin = declare((api, options: Options) => {
           }
         }
       },
+      ImportDeclaration(path: NodePath<t.ImportDeclaration>) {
+        if (options.auto?.includes(path.node.source.value)) {
+          // get specifiers:
+          path.node.specifiers.forEach((specifier) => {
+            // rename specifier to _specifier
+            // and insertAfter const specifier = block(_specifier)
+            path.scope.rename(specifier.local.name, `_${specifier.local.name}`);
```
Comment (poteboy) [inline packages/compiler/plugin.ts:115]: Does this change only apply to named exports?

File: packages/compiler/react/utils.ts [hunk 2]
```diff
@@ -1,7 +1,69 @@
...
+import { parseSync, traverse } from '@babel/core';
+import { getBasePlugins, getIndex } from '../plugin';
 import type { NodePath } from '@babel/core';
 import type { Options } from '../plugin';
 
+export const getSpecifierSource = (
+  importSpecifierPath: NodePath<t.ImportSpecifier>,
+  id: string,
+): NodePath<t.FunctionDeclaration | t.VariableDeclaration> | null => {
+  const source = resolvePath(importSpecifierPath.parentPath.get('source')).node;
```
Comment (poteboy) [inline packages/compiler/react/utils.ts:13]: Ah, I wasn't aware this could work

File: packages/compiler/react/visitor.ts [hunk 3]
```diff
@@ -180,27 +186,34 @@ export const visitor = (options: Options = {}, isReact = true) => {
...
     )!.path;
 
+    if (componentDeclarationPath.isImportSpecifier()) {
+      const correctComponentDeclarationPath = getSpecifierSource(
+        componentDeclarationPath,
+        options.id!,
+      );
+      if (correctComponentDeclarationPath) {
+        componentDeclarationPath.remove();
+        globalPath.insertBefore(correctComponentDeclarationPath.node);
```
Comment (poteboy) [inline packages/compiler/react/visitor.ts:203]: What happens when the imported component depends on external scope variables in the original file? For instance:
```tsx
// in original.tsx
const externalVariable = 'Hello';
export const Component = () => <div>{externalVariable}</div>;

// in different-file.tsx
import { Component } from './original';
const ComponentBlock = block(Component);
```

**Comment by tobySolutions** [comment]
Hmm, good work on this @aidenybai, thank you for the review @poteboy!

**Comment by tobySolutions** [comment]
The work here looks really interesting. I hope it is completed soon though.

**Pull request closed without merging**